Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java

```diff
@@ -0,0 +1,124 @@
+package org.gusdb.wdk.model.fix.table.edaanalysis.plugins;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.log4j.Logger;
+import org.gusdb.fgputil.json.JsonUtil;
+import org.gusdb.wdk.model.WdkModel;
+import org.gusdb.wdk.model.fix.table.TableRowInterfaces;
+import org.gusdb.wdk.model.fix.table.edaanalysis.AbstractAnalysisUpdater;
+import org.gusdb.wdk.model.fix.table.edaanalysis.AnalysisRow;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class VDIMigrationPlugin extends AbstractAnalysisUpdater {
+  private static final Logger LOG = Logger.getLogger(VDIMigrationPlugin.class);
+  public static final String UD_DATASET_ID_PREFIX = "EDAUD_";
+
+  private Map<String, String> legacyIdToVdiId;
+  private int missingFromVdiCount = 0;
+
+  @Override
+  public TableRowInterfaces.RowResult<AnalysisRow> processRecord(AnalysisRow nextRow) throws Exception {
+    final String legacyDatasetId = nextRow.getDatasetId();
+    final String legacyUdId = legacyDatasetId.replace(UD_DATASET_ID_PREFIX, "");
+    final String vdiId = legacyIdToVdiId.get(legacyUdId);
+
+    if (vdiId == null) {
+      LOG.warn("Unable to find legacy ID " + legacyUdId + " in the tinydb file.");
+      missingFromVdiCount++;
+      return new TableRowInterfaces.RowResult<>(nextRow);
+    }
+
+    // Append UD prefix to VDI ID. The prefix is prepended in the view that maps stable VDI IDs to the unstable study
+    // ID, which is the currency of EDA.
+    final String vdiDatasetId = UD_DATASET_ID_PREFIX + vdiId;
+
+    // Create a copy with just the dataset ID updated to VDI counterpart.
+    AnalysisRow out = new AnalysisRow(nextRow.getAnalysisId(), vdiDatasetId, nextRow.getDescriptor(),
+        nextRow.getNumFilters(), nextRow.getNumComputations(), nextRow.getNumVisualizations());
+
+    return new TableRowInterfaces.RowResult<>(out);
```

Comment:
Want to set the result object's update flag to true here unless _writeToDb is false.